### PR TITLE
governance: Add GOVERNANCE.md document

### PR DESCRIPTION
**Pull request by mestery**
This commit adds an initial stab at a Diamond Bluff governance
structure. This includes both a Board of Directors, as well as a
Technical Steering Commmittee.

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

File: GOVERNANCE.md
Comment (jberkus) [inline GOVERNANCE.md:83]: So by this formulation, the majority of the members of the TOC would also be direct appointees, rather than representing the projects.  This makes the TOC not at all differentiated from the Board.  We need a structure that makes sure that we get a different group of people for the TOC; let me see if I can add some text to make that happen.

Comment (mestery) [reply GOVERNANCE.md:83]: Excellent and yes, I want the Technical Steering Committee (TSC) to be a bit more technical and less appointed, I can put some text to change that if you can't push to this PR.

Comment (jberkus) [reply GOVERNANCE.md:83]: What's the largest number of projects we expect to have under DB?

Comment (mestery) [reply GOVERNANCE.md:83]: That's hard to predict, but this governance document is trying to take a first crack at how those might be formed and guided by the Technical Steering Committee. I can see 3 initial projects: A DPU/IPU infra project (think base OS and packaging), IPDK (table driver interface abstractions), and testing.

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:19]: What about individuals? those who are not aligned with a company, nor with an academic institution. I'm not sure he will continue work with Diamond bluff, but Dave Schmitt is a prime example. He just retired from F5, but may want to be involved.

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:24]:
```suggestion
Platinum, Silver, Individual, and Participating Academic Members shall be entitled to:
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:28]:
```suggestion
* Identify their company or academic institution as a member or participant in the Diamond Bluff
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:55]:
```suggestion
  authorize expenditures approved by the budget, manage any day-to-day
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:69]:
```suggestion
* Engage the project's foundation for all press and analyst
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:64]:
```suggestion
  community, while also maintaining the goodwill and esteem of the 
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:73]:
```suggestion
 project's foundation.
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:74]:
```suggestion
* Engage the project's foundation for any Diamond Bluff project related websites.
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:75]:
```suggestion
and operate under such rules and procedures as may
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:77]:
```suggestion
  confirmed by the Project's foundation.
```

File: GOVERNANCE.md
Comment (pdp2shirts) [reply GOVERNANCE.md:83]: I thought that we were going to adopt something along the lines of an appointed TSC member from each platinum member to start with, and have 1/3 rotate off each of the first 3 years, to make way for TSC members to be nominated and voted in on technical contribution merit. that way we have a TSC to begin with that will have some consistency over the beginning years, but end up with those of great technical merit in the end.

Comment (pdp2shirts) [reply GOVERNANCE.md:83]: I do not have a problem adding a maintainer from each subproject either.

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:94]:
```suggestion
* In coordination with the Board of Directors, approving new sub-projects for
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:109]:
```suggestion
Decisions by vote shall be based on a majority vote only when sixty percent or greater
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:110]:
```suggestion
(>= 60%) of the Board of Directors representatives are either present or
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:117]:
```suggestion
in excess of funds raised. The budget shall be consistent with the non-profit
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:118]:
```suggestion
mission of the Diamond Bluff project's foundation. The foundation shall
```

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:119]: can we commit the foundation to provide those reports?

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:124]:
```suggestion
of. A super majority vote is defined as a vote in favor by 70% or greater of the
```

**Review by pdp2shirts — CHANGES_REQUESTED** [review]

File: GOVERNANCE.md
Comment (pdp2shirts) [inline GOVERNANCE.md:26]:
```suggestion
* Participate in the Diamond Bluff Project meetings, initiatives, events and
```

File: GOVERNANCE.md
Comment (jberkus) [reply GOVERNANCE.md:82]: Yeah, that's what I recall too.

**Comment by jberkus** [comment]
BTW, my thinking is that we should just keep this a draft until we select a foundation.  The foundation will have feedback on what kind of structure their folks can help administer.

File: GOVERNANCE.md
Comment (jberkus) [inline GOVERNANCE.md:103]: Should this voting section refer to the TSC as well as the Board?

File: GOVERNANCE.md
Comment (jberkus) [inline GOVERNANCE.md:78]: Alternate version:

### Technical Steering Committee

Members of the Diamond Bluff Technical Steering Committee (TSC) decide technical 
matters that affect the entire project, administer joint project-wide 
technical resources, and act as a final arbiter of technical matters that are 
not decided inside individual subprojects.

#### TSC Membership

All TSC members must be currently contributing engineers, documentation writers,
test platform adminstrators, specification editors, or other direct technical 
contributors to one or more subprojects.

During the first year of of the project, The TSC shall be composed as follows:

* One appointed representative from each Platinum member
* One maintainer from each Diamond Bluff sub-project, selected by the Board

Starting one year after the launch of the Diamond Bluff governance, the TSC 
composition will become:

* Two maintainers from each Diamond Bluff sub-project
* Two technical contributors appointed by the Board

Membership in the TSC will gradually transition to a model of elected subproject
technical leadership, starting in the second year of the project. Details for 
this transition will be decided by the Board.

Should any TSC member cease regular technical contributions, their membership
will terminate at the next regular annual selection date. If a TSC member
violates the Code of Conduct, they may be removed from the TSC by the Code
of Conduct Committee.

#### TSC Responsibilities

The responsibilities of the Diamond Bluff Technical Steering Committee are
as follows:

* Coordinating the technical direction of the Diamond Bluff Project
* Nominating and reviewing new sub-projects for creation under the Diamond Bluff 
  organization after final approval by the Board.
* Confirming selected maintainers in the sub-projects, and ensuring they have the
  technical support necessary to be successful in their role as a maintainer.
* Communicating with external and industry organizations concering Diamond
  Bluff technical matters.
* Appointing represenatives to work with other open source or open standards
  communities.
* Voting on technical matters relating to any common code base or specifications
  affecting all sub-projects.
* Adminstering common technical resources, such as repository hosting and the 
  testing infrastructure.
* Acting as "court of final appeal" for technical matters which are not resolved
  through normal decision mechanisms within the individual subprojects.
* Coordinating response to security issues and ensuring that all subprojects
  are reasonably secure.
  
For any of the above responsibilities, the TSC may delegate their powers to one
or more parties as appropriate and necessary, under terms set by the TSC.

#### TSC Meetings

The TSC shall hold one or more public meetings each month.  All contributors
to, or sponsors of, Diamond Bluff shall be entitled to attend these meetings. 
Public meetings will have a public agenda to which members of the community may
reasonably add items.

The TSC may also have private meetings from time to time, particularly when
discussing security response or settling escalated disputes. All TSC members
will be included in private meetings.

Comment (jberkus) [reply GOVERNANCE.md:78]: This version does several things:  

1. Ensures that the TSC is made up of people actually working on the subprojects
2. Adds the expectation that the TSC will transition to a model of technical promotion from below instead of appointment by the Board, but leaves that exact model flexible.
3. Adds a more complete set of TSC responsibilities.

File: GOVERNANCE.md
Comment (jberkus) [inline GOVERNANCE.md:70]: Not sure why this needs to be in there?  Won't the foundation have more info about events than we will?

File: GOVERNANCE.md
Comment (jberkus) [inline GOVERNANCE.md:76]: Add:

For any of the above responsibilities, the Board may delegate their powers to one
or more parties as appropriate and necessary, under terms set by the Board.

File: GOVERNANCE.md
Comment (jberkus) [inline GOVERNANCE.md:48]: ... every 5 or fraction thereof

In other words, when we get 6 silver members, we'll elect 2 silver reps

File: GOVERNANCE.md
Comment (jberkus) [inline GOVERNANCE.md:111]: Add para:

All elections shall be held using an online preference election mechanism to be selected and approved by the Board.

File: GOVERNANCE.md
Comment (jberkus) [inline GOVERNANCE.md:102]: Add section:

## Code of Conduct Committee

The Board will appoint or elect three or more individuals to serve as a Code Of Conduct Committee (COCC).  Committee members may also be members of the Board or the TSC, or they may be outside parties (paid or unpaid) with expertise in code of conduct (CoC) enforcement, or just members of the community.  The CoCC should be selected to represent the diversity of the community.

The CoCC will establish fair and standard processes for handling reports of violations of the CoC, as well as a system of remedies, in order to protect the well-being of the project.  The CoCC will recommend the appropriate resolution to the Board, up to and including the removal of any individual from participation, either temporarily or permanently.  Should a member of the CoCC itself be accused of a violation, that accusation will be handled by an individual appointed by the foundation.

**Comment by mestery** [comment]
> BTW, my thinking is that we should just keep this a draft until we select a foundation. The foundation will have feedback on what kind of structure their folks can help administer.

My thinking is that we should decide the governance we want, and have the Foundation set that up for us, not that we should take whatever governance they want to manage.

File: GOVERNANCE.md
Comment (mestery) [reply GOVERNANCE.md:70]: Superficial, yes. But a deeper analysis, likely not as much as the maintainers and participants.

File: GOVERNANCE.md
Comment (mestery) [reply GOVERNANCE.md:78]: I like this version, thanks for the help!

File: GOVERNANCE.md
Comment (mestery) [reply GOVERNANCE.md:98]: The latest version is updated to reflect this discussion.

**Comment by mestery** [comment]
> BTW, my thinking is that we should just keep this a draft until we select a foundation. The foundation will have feedback on what kind of structure their folks can help administer.

I'd actually like to merge this and use this as what we want the foundation to enable for us, rather than letting them dictate it to us. We're paying them, after all, so they should be doing what the community wants.

In this regard, can you review this again now that I've incorporated your comments and feedback? Thanks!

**Review by pdp2shirts — COMMENTED** [review]
this looks good. a few typos and suggestions for clarifying language at line 104

File: GOVERNANCE.md
Comment (mestery) [reply GOVERNANCE.md:108]: Will update throughout the doc to be consistently `sub-project`.

**Pull request merged**
